server/src/main/java/org/elasticsearch/rest/action/ingest/RestPutPipelineAction.java

```diff
@@ -47,7 +48,8 @@ public String getName() {
     @Override
     public RestChannelConsumer prepareRequest(RestRequest restRequest, NodeClient client) throws IOException {
         Tuple<XContentType, BytesReference> sourceTuple = restRequest.contentOrSourceParam();
-        PutPipelineRequest request = new PutPipelineRequest(restRequest.param("id"), sourceTuple.v2(), sourceTuple.v1());
+        PutPipelineRequest request = new PutPipelineRequest(restRequest.param("id"),
```

Comment:
As mentioned in the comments, can't escape the bytes here (these are held on to by the ingest logic internally) since they'll be released eventually so we copy.